torchaudio/functional/filtering.py

```diff
@@ -642,6 +642,48 @@ def equalizer_biquad(
     return biquad(waveform, b0, b1, b2, a0, a1, a2)
 
 
+def filtfilt(
+    waveform: Tensor,
+    a_coeffs: Tensor,
+    b_coeffs: Tensor,
+    clamp: bool = True,
+    batching: bool = True,
+) -> Tensor:
+    r"""Apply an IIR filter forward and backward to a waveform.
+
+    Inspired by https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.filtfilt.html
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of ``(..., time)``.  Must be normalized to -1 to 1.
+        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of either
+                                1D with shape ``(num_order + 1)`` or 2D with shape ``(num_filters, num_order + 1)``.
+                                Lower delay coefficients are first, e.g. ``[a0, a1, a2, ...]``.
+                                Must be same size as b_coeffs (pad with 0's as necessary).
+        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of either
+                                1D with shape ``(num_order + 1)`` or 2D with shape ``(num_filters, num_order + 1)``.
+                                Lower delay coefficients are first, e.g. ``[b0, b1, b2, ...]``.
+                                Must be same size as a_coeffs (pad with 0's as necessary).
+        clamp (bool, optional): If ``True``, clamp the output signal to be in the range [-1, 1] (Default: ``True``)
+        batching (bool, optional): Effective only when coefficients are 2D. If ``True``, then waveform should be at
```

Comment:
I personally would like not to expose `batching` option for now. (and pass `batching=True` to `lfilter`)
`batching=False` option in `lfilter` is very unique feature, not following the general convention of DL frameworks. So first I would like to gauge the popularity/usefulness in `lfilter` then adopt it to the other functions. From software perspective, it's easy to add a public interface but very difficult to remove it.

@yoyololicon What do you think?

Reply:
@mthrok 
`lfilter` and `filtfilt` are very similar functions so it looks natural to me to sync the changes together.
But I understand your concern,  I'm ok with not adopt this parameter on `filtfilt` first.
I do consider extend the batching/filterbanks feature on biquad filters as well in the future.